### PR TITLE
Feature/aw/busy UI test

### PR DESCRIPTION
**Pull request by 6Botter7**
Animation added and Progress text

### CODE REVIEW

File: components/busy-ui/busy-ui.html [hunk 1]
```diff
@@ -7,9 +7,75 @@
...
+        text-align: center;
     }
+
+    .loading span {
+        width: 1rem;
+        height: 1rem;
+        background: var(--blue-s1);
+        border-radius: 100%;
+        display: inline-block;
+        -webkit-animation: dots 1.5s infinite ease-in-out both;
```
Comment (Ashton-Pragma) [inline components/busy-ui/busy-ui.html]: check if current firefox still needs this

File: components/busy-ui/busy-ui.html [hunk 2]
```diff
@@ -7,9 +7,75 @@
...
+    }
+
+    #lblProgress {
+        text-align: center;
+        margin-bottom: 0.5rem;
+        font-weight: normal;
+        color: var(--grey-s1);
+        font-style: italic;
+        font-size: 1rem;
+        font-family: 'SourceSansPro', sans-serif;
```
Comment (Ashton-Pragma) [inline components/busy-ui/busy-ui.html]: do we need this? can it be imported ?

File: components/busy-ui/busy-ui.js [hunk 3]
```diff
@@ -47,10 +49,27 @@ export class BusyUi extends HTMLElement {
...
     }
+
+    /**
+     * @method attributeChangedCallback - called when an attribute changes
+     * @return {Promise<void>}
+     */
+    async attributeChangedCallback(name, newVal, oldVal) {
+        const lblMessage = this.shadowRoot.querySelector("#lblMessage");
+        const lblProgress = this.shadowRoot.querySelector("#lblProgress");
+        if (lblMessage) {
```
Comment (Ashton-Pragma) [inline components/busy-ui/busy-ui.js]: be explicit == null or != null

**Review by Ashton-Pragma — CHANGES_REQUESTED** [review]

File: components/busy-ui/busy-ui.html (hunk 1 above)
Comment (6Botter7) [reply components/busy-ui/busy-ui.html]: added -moz for older versions, newer versions from September 2021, both Chrome and Firefox support unprefixed keyframes and transformations.

File: components/busy-ui/busy-ui.html (hunk 2 above)
Comment (6Botter7) [reply components/busy-ui/busy-ui.html]: removed size and font-family, italic is a design requirement

File: components/busy-ui/busy-ui.js (hunk 3 above)
Comment (6Botter7) [reply components/busy-ui/busy-ui.js]: Added explicit null checks

File: components/busy-ui/busy-ui.js [hunk 4]
```diff
@@ -14,6 +14,14 @@
  * <busy-ui data-message="hello world"></busy-ui>
  */
 export class BusyUi extends HTMLElement {
+    /**
+     * @method observedAttributes - the attributes to watch for changes
+     * @return {[string]}
+     */
+    static get observedAttributes() {
+        return ['data-progress'];
```
Comment (Ashton-Pragma) [inline components/busy-ui/busy-ui.js]: you also need to have data-message in here otherwise it wont trigger

**Review by Ashton-Pragma — CHANGES_REQUESTED** [review]

File: components/busy-ui/busy-ui.js (hunk 4 above)
Comment (6Botter7) [reply components/busy-ui/busy-ui.js]: data-message added.

**Review by Ashton-Pragma — APPROVED** [review]

**Pull request merged**